client/extensions/woocommerce/state/data-layer/request/index.js

```diff
@@ -20,6 +20,12 @@ export function handleRequest( { dispatch }, action ) {
 				action,
 				data,
 			} );
+
+			if ( action.onSuccessAction ) {
+				// Append data and dispatch.
+				const onSuccess = { ...action.onSuccessAction, data };
```

Comment:
I see your point. I guess the intermediate variable isn't really getting us much, here.